[Style] Convert remaining integer/number properties to strong style types

weinig opened the pull request:
#### c59786251fb888b7a9c9b545967f8a0b299d9426
<pre>
[Style] Convert remaining integer/number properties to strong style types
<a href="https://bugs.webkit.org/show_bug.cgi?id=297059">https://bugs.webkit.org/show_bug.cgi?id=297059</a>

Reviewed by Darin Adler.

Convert the following remaining integer/number properties to strong style types:
    - `column-count`
    - `column-width`
    - `flex-grow`
    - `flex-shrink`
    - `order`
    - `z-index`

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/inspector/InspectorOverlay.cpp:
* Source/WebCore/layout/formattingContexts/flex/FlexFormattingContext.cpp:
* Source/WebCore/layout/formattingContexts/flex/FlexLayout.cpp:
* Source/WebCore/layout/formattingContexts/flex/LogicalFlexItem.h:
* Source/WebCore/page/ElementTargetingController.cpp:
* Source/WebCore/page/InteractionRegion.cpp:
* Source/WebCore/rendering/OrderIterator.cpp:
* Source/WebCore/rendering/RenderBlockFlow.cpp:
* Source/WebCore/rendering/RenderBox.cpp:
* Source/WebCore/rendering/RenderElement.cpp:
* Source/WebCore/rendering/RenderFlexibleBox.cpp:
* Source/WebCore/rendering/RenderLayer.cpp:
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
* Source/WebCore/rendering/RenderLayerInlines.h:
* Source/WebCore/rendering/RenderTreeAsText.cpp:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleBoxData.cpp:
* Source/WebCore/rendering/style/StyleBoxData.h:
* Source/WebCore/rendering/style/StyleFlexibleBoxData.h:
* Source/WebCore/rendering/style/StyleMiscNonInheritedData.h:
* Source/WebCore/rendering/style/StyleMultiColData.cpp:
* Source/WebCore/rendering/style/StyleMultiColData.h:
* Source/WebCore/rendering/updating/RenderTreeBuilderMultiColumn.cpp:
* Source/WebCore/style/StyleAdjuster.cpp:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleResolveForDocument.cpp:
* Source/WebCore/style/values/css2/StyleZIndex.cpp: Added.
* Source/WebCore/style/values/css2/StyleZIndex.h: Added.
* Source/WebCore/style/values/display/StyleOrder.h: Added.
* Source/WebCore/style/values/flexbox/StyleFlexGrow.h: Added.
* Source/WebCore/style/values/flexbox/StyleFlexShrink.h: Added.
* Source/WebCore/style/values/multicol/StyleColumnCount.cpp: Added.
* Source/WebCore/style/values/multicol/StyleColumnCount.h: Added.
* Source/WebCore/style/values/multicol/StyleColumnWidth.cpp: Added.
* Source/WebCore/style/values/multicol/StyleColumnWidth.h: Added.

Canonical link: <a href="https://commits.webkit.org/298473@main">https://commits.webkit.org/298473@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/43071e97649276a2dcb025311683f86264693e10

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115583 "5 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35246 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25769 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121630 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66101 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117472 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35927 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43829 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87796 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42437 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118531 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28653 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103731 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68194 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27790 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21846 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65309 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98027 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21968 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124783 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42478 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31839 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96563 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42845 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99921 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96349 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24524 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41593 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19451 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38354 "Hash 43071e97 for PR 49067 does not build (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42369 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47941 "Built successfully") | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41847 "Built successfully") | | [⏳ 🧪 jsc-armv7-tests ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45179 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43562 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fa04907c4590c69cc6e2280c7be3bf34f62dfd0a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49067

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e814e3d781602612623b63e4eb827f46a58e810d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49067

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d5f412b0a3572cc4309d68abc146afbaa22b91c1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49067

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/43071e97649276a2dcb025311683f86264693e10. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49067

weinig committed:
[Style] Convert remaining integer/number properties to strong style types
https://bugs.webkit.org/show_bug.cgi?id=297059

Reviewed by Darin Adler.

Convert the following remaining integer/number properties to strong style types:
    - `column-count`
    - `column-width`
    - `flex-grow`
    - `flex-shrink`
    - `order`
    - `z-index`

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/inspector/InspectorOverlay.cpp:
* Source/WebCore/layout/formattingContexts/flex/FlexFormattingContext.cpp:
* Source/WebCore/layout/formattingContexts/flex/FlexLayout.cpp:
* Source/WebCore/layout/formattingContexts/flex/LogicalFlexItem.h:
* Source/WebCore/page/ElementTargetingController.cpp:
* Source/WebCore/page/InteractionRegion.cpp:
* Source/WebCore/rendering/OrderIterator.cpp:
* Source/WebCore/rendering/RenderBlockFlow.cpp:
* Source/WebCore/rendering/RenderBox.cpp:
* Source/WebCore/rendering/RenderElement.cpp:
* Source/WebCore/rendering/RenderFlexibleBox.cpp:
* Source/WebCore/rendering/RenderLayer.cpp:
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
* Source/WebCore/rendering/RenderLayerInlines.h:
* Source/WebCore/rendering/RenderTreeAsText.cpp:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleBoxData.cpp:
* Source/WebCore/rendering/style/StyleBoxData.h:
* Source/WebCore/rendering/style/StyleFlexibleBoxData.h:
* Source/WebCore/rendering/style/StyleMiscNonInheritedData.h:
* Source/WebCore/rendering/style/StyleMultiColData.cpp:
* Source/WebCore/rendering/style/StyleMultiColData.h:
* Source/WebCore/rendering/updating/RenderTreeBuilderMultiColumn.cpp:
* Source/WebCore/style/StyleAdjuster.cpp:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleResolveForDocument.cpp:
* Source/WebCore/style/values/css2/StyleZIndex.cpp: Added.
* Source/WebCore/style/values/css2/StyleZIndex.h: Added.
* Source/WebCore/style/values/display/StyleOrder.h: Added.
* Source/WebCore/style/values/flexbox/StyleFlexGrow.h: Added.
* Source/WebCore/style/values/flexbox/StyleFlexShrink.h: Added.
* Source/WebCore/style/values/multicol/StyleColumnCount.cpp: Added.
* Source/WebCore/style/values/multicol/StyleColumnCount.h: Added.
* Source/WebCore/style/values/multicol/StyleColumnWidth.cpp: Added.
* Source/WebCore/style/values/multicol/StyleColumnWidth.h: Added.

Canonical link: https://commits.webkit.org/298473@main
